src/Microsoft.Health.Dicom.Api/Controllers/CustomTagController.cs

```diff
@@ -50,5 +51,51 @@ public async Task<IActionResult> PostAsync([FromBody] IEnumerable<CustomTagEntry
             return StatusCode(
                (int)HttpStatusCode.Accepted, response);
         }
+
+        /// <summary>
+        /// Handles requests to get all custom tags.
+        /// </summary>
+        /// <returns>
+        /// Returns Bad Request if given path can't be parsed. Returns Not Found if given path doesn't map to a stored
+        /// custom tag or if no custom tags are stored. Returns OK with a JSON body of all tags in other cases.
+        /// </returns>
+        [ProducesResponseType(typeof(JsonResult), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [HttpGet]
+        [Route(KnownRoutes.CustomTagRoute)]
+        [AuditEventType(AuditEventSubType.GetAllCustomTags)]
+        public async Task<IActionResult> GetAllTagsAsync()
+        {
+            _logger.LogInformation("DICOM Web Get Custom Tag request received for all custom tags");
+
+            GetCustomTagResponse response = await _mediator.GetCustomTagsAsync(null, HttpContext.RequestAborted);
+
+            return StatusCode(
+                (int)HttpStatusCode.OK, response);
+        }
+
+        /// <summary>
+        /// Handles requests to get individual custom tags.
+        /// </summary>
+        /// <param name="tagPath">Path for requested custom tag.</param>
+        /// <returns>
+        /// Returns Bad Request if given path can't be parsed. Returns Not Found if given path doesn't map to a stored
+        /// custom tag. Returns OK with a JSON body of requested tag in other cases.
+        /// </returns>
+        [ProducesResponseType(typeof(JsonResult), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [HttpGet]
+        [Route(KnownRoutes.GetCustomTagRoute)]
+        [AuditEventType(AuditEventSubType.GetCustomTag)]
+        public async Task<IActionResult> GetTagAsync(string tagPath)
+        {
+            _logger.LogInformation("DICOM Web Get Custom Tag request received for custom tag: {0}", tagPath);
```

Comment:
nit: instead of `{0}`, you should be able to do `{tagPath}` so that the name is carried through.